src/mongo/client/dbclient.cpp

```diff
@@ -1365,6 +1366,13 @@ void DBClientWithCommands::_findAndModify(const StringData& ns,
     commandBuilder.append("new", returnNew);
     commandBuilder.append("upsert", upsert);
 
+    if (getMaxWireVersion() >= 4) {
```

Comment:
if the maxWireVersion is too low, should we uassert that the operation is not supported on the remote server?